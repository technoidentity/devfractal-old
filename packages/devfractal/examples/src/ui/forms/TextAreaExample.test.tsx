import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'
import { TextArea } from './TextAreaExample'

test('TextArea Example', () => {
  const component: ReactTestRenderer = renderer.create(<TextArea />)
  expect(component.toJSON()).toMatchSnapshot()
})
