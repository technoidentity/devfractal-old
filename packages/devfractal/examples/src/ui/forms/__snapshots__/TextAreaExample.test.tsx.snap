// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea Example 1`] = `
Array [
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Dynamic Textarea(Upper case)
    </h1>
    <div
      className="control"
    >
      <textarea
        className="textarea"
        name="message"
        onChange={[Function]}
        placeholder="Dynamic Textarea"
        value=""
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Default Textarea
    </h1>
    <div
      className="control"
    >
      <textarea
        className="textarea"
        placeholder="Default textArea"
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Colors
    </h1>
    <div
      className="control"
    >
      <textarea
        className="textarea is-primary"
        placeholder="Primary textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-info"
        placeholder="Info textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-success"
        placeholder="Success textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-warning"
        placeholder="Warning textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-danger"
        placeholder="Danger textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-light"
        placeholder="Light textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-dark"
        placeholder="Dark textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-black"
        placeholder="Black textarea"
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Sizes
    </h1>
    <div
      className="control is-small"
    >
      <textarea
        className="textarea is-small"
        placeholder="Small textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea"
        placeholder="Normal textarea"
      />
    </div>
    <div
      className="control is-medium"
    >
      <textarea
        className="textarea is-medium"
        placeholder="Medium textarea"
      />
    </div>
    <div
      className="control is-large"
    >
      <textarea
        className="textarea is-large"
        placeholder="Large textarea"
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      States
    </h1>
    <div
      className="control"
    >
      <textarea
        className="textarea"
        placeholder="Normal textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-active"
        placeholder="Active textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-hovered"
        placeholder="Hovered textarea"
      />
    </div>
    <div
      className="control"
    >
      <textarea
        className="textarea is-focused"
        placeholder="Focused textarea"
      />
    </div>
    <div
      className="control is-loading"
    >
      <textarea
        className="textarea"
        placeholder="Loading textarea"
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Resize Loading Spinner
    </h1>
    <div
      className="control is-loading is-small"
    >
      <textarea
        className="textarea is-small"
        placeholder="Small loading textarea"
      />
    </div>
    <div
      className="control is-loading"
    >
      <textarea
        className="textarea"
        placeholder="Normal loading textarea"
      />
    </div>
    <div
      className="control is-loading is-medium"
    >
      <textarea
        className="textarea is-medium"
        placeholder="Medium loading textarea"
      />
    </div>
    <div
      className="control is-loading is-large"
    >
      <textarea
        className="textarea is-large"
        placeholder="Large loading textarea"
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Fixed Size
    </h1>
    <div
      className="control"
    >
      <textarea
        className="textarea has-fixed-size"
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Disabled
    </h1>
    <div
      className="control"
    >
      <textarea
        className="textarea"
        disabled={true}
        placeholder="Disabled textarea"
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Height
    </h1>
    <div
      className="control"
    >
      <textarea
        className="textarea"
        rows={10}
      />
    </div>
  </section>,
  <section
    className="section"
  >
    <h1
      className="title is-4"
    >
      Readonly
    </h1>
    <div
      className="control"
    >
      <textarea
        className="textarea"
        readOnly={true}
        value="The content is Readonly"
      />
    </div>
  </section>,
]
`;
